errors/panic_handler.go

```diff
@@ -0,0 +1,18 @@
+package errors
+
+import (
+	"context"
+)
+
+// PanicHandler is the interface used to create custom panic errors that occur during query execution
+type PanicHandler interface {
+	MakePanicError(ctx context.Context, value interface{}) *QueryError
+}
```

Comment:
The interface needs to be declared where it is used, not where you declare implementations of that interface. Please, move the interface, where it is used.

Reply:
OK, I followed the existing pattern for `log.Logger` which has the interface defined in the same file as the default logger. But thanks for the pointer, will move the `PanicHandler` interface to the `exec.go` file.